WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift

```diff
@@ -53,7 +54,7 @@ final class POSEligibilityChecker: POSEligibilityCheckerProtocol {
 }
 
 private extension POSEligibilityChecker {
-    var isWooCommerceVersionSupported: AnyPublisher<Bool, Never> {
+    var isWooCommerceVersionSupportedAndFeatureSwitchEnabled: AnyPublisher<Bool, Never> {
```

Comment:
Should we keep these checks separate and happen in parallel?

Reply:
Maybe we could even ditch Combine and use async/await here, and async let for parallel requests?

Reply:
Yes, the code is readable. I agree it's not pretty, but that's fine. `POSEligibilityChecker` is not expected to get much more complicated.

Reply:
Thanks for the feedback, I'll merge this version.

> Maybe we could even ditch Combine and use async/await here, and async let for parallel requests?

Yea, I was trying to think why we didn't start with async/await. I think it was because at the time, we're still checking the payments onboarding state, which could change from the payments tab or other parts of the app and we wanted to observe the state and update the POS eligibility in real-time.

Now that we don't have this onboarding state requirement anymore, I think we can refactor it to async/await for better readability. Maybe when we're making more changes on the POS eligibility in the future.